scripts/metrics/common/query_metrics.py

```diff
@@ -0,0 +1,167 @@
+    :param desired_branch: name of the desired branch
+    :param start_week: number of the first week to include
+    :param end_week: number of the last week to include
+    :param download_folder: destination folder for the .csvs to be stored while they are being parsed
+    :param region: desired AWS region for session
+    :param use_env_credentials: use the env credentials instead of the configured boto3 credentials
+    """
+    bucket = _get_session(region, ).resource('s3').Bucket(BUCKET_NAME) if use_env_credentials else boto3.client('s3')
+
+    prefix = "csv/" + desired_branch
+
+    objects = list(bucket.objects.filter(Prefix=prefix))
+
+    file_type = ".csv"
+    for object in objects:
+        if file_type == os.path.splitext(object.key)[1]:
+            try:
+                # Get the key file values
+                # We are expecting files to be found in buckets/o3de-metrics/csv/branch/Week#/file_name.csv
+                # If they are not, they won't be checked, and will be caught in the ValueError below
+                filter_values = object.key.split('/')
+                found_branch = filter_values[-3]
+                week_number = int((filter_values[-2])[4:])
+                file_name = filter_values[-1]
+                print("found_branch: " + str(found_branch))
+                print("week_number: " + str(week_number))
+                print("file_name: " + str(file_name))
+
+                # Determine if the file should be downloaded
+                should_download_file = True
+                if found_branch != desired_branch:
+                    should_download_file = False
+                if start_week and week_number < start_week:
+                    should_download_file = False
+                elif end_week and week_number > end_week:
+                    should_download_file = False
+
+                # Download the file
+                if should_download_file:
+                    print("str(download_folder): " + str(download_folder))
+                    print("str(week_number): " + str(week_number))
+                    week_folder = os.path.join(download_folder, str(week_number))
+                    if not os.path.exists(week_folder):
+                        os.mkdir(week_folder)
+                    download_path = os.path.join(download_folder, str(week_number), file_name)
+                    DOWNLOADED_FILES.append(download_path)
+                    bucket.download_file(object.key, download_path)
+
+            # If a .csv is in the wrong location, it'll break the functionality above. We're just going to skip it.
+            except ValueError as value_error:
+                print("Found value error. Likely means a csv is in the wrong location. Skipping and continuing. Error "
+                      "{0} ".format(str(value_error)))
+
+
+def query_csvs(test_case_ids=None, test_case_metrics=None):
+    """
+    Queries the desired information from the downloaded csvs
```

Comment:
I'm not quite sure what I would use this function for in the future. I expect any report generating tools will be using tools like numpy to interact with CSV file data and matplotlib or seaborn to make any graphs.

Reply:
Still proves the CSVs exist and can be interacted with.